### ISSUE TITLE
Launch Settings not detecting source code projects correctly

### ISSUE THREAD

**Issue by WolfspiritM** [microsoft/DockerTools#303]
Hi,

We have many long project names similar to:
Company.Services.WebApp

This creates a docker-compose service called "company.services.webapp".

However according to docker dots are not allowed in service names: https://github.com/docker/swarmkit/issues/2437#issuecomment-343573663

Therefor we changed our service name to "webapp" so we can deploy it in docker swarm using the same docker-compose.
And also this way we can reference it as "http://webapp" from other containers.

However now when opening the "Manage Docker Compose Launch Settings" there is no "Start debugging" anymore.

I think either projects should be able to define their "DockerComposeServiceName" in the csproj file or there should be some other mechanism to detect if it's a project with sourcecode (maybe by using the context instead of the name)

**Comment by NCarlsonMSFT** [comment]
@WolfspiritM I took a quick look, and this appears to be a bug in the UI for the launch settings. You can work around this by opening launchsettings.json and editing the file. For example I stared with a project named Foo.Bar, renamed the service to foo-bar and was able to enable debugging with:
```json
{
  "profiles": {
    "Docker Compose": {
      "commandName": "DockerCompose",
      "composeLaunchServiceName": "foo-bar",
      "serviceActions": {
        "foo-bar": "StartDebugging",
      },
      "composeLaunchAction": "LaunchBrowser",
      "commandVersion": "1.0",
      "composeLaunchUrl": "{Scheme}://localhost:{ServicePort}"
    }
  }
}
```

**Comment by NCarlsonMSFT** [comment]
@WolfspiritM I took a quick look, and this appears to be a bug in the UI for the launch settings. You can work around this by opening launchsettings.json and editing the file. For example I stared with a project named Foo.Bar, renamed the service to foo-bar and was able to enable debugging with:
```j
{
  "profiles": {
    "Docker Compose": {
      "commandName": "DockerCompose",
      "composeLaunchServiceName": "foo-bar",
      "serviceActions": {
        "foo-bar": "StartDebugging",
      },
      "composeLaunchAction": "LaunchBrowser",
      "commandVersion": "1.0",
      "composeLaunchUrl": "{Scheme}://localhost:{ServicePort}"
    }
  }
}
```

**Comment by ravipal** [comment]
@WolfspiritM This is a known issue  when the service name is different than project name, then the 'start debugging' options is not shown in the UI.  This has been fixed in the latest preview release 16.11 Preview 3 which can be downloaded from https://visualstudio.microsoft.com/vs/preview/.
